src/neo-vm/Unsafe.cs

```diff
@@ -36,6 +36,10 @@ public static bool MemoryEquals(byte[] x, byte[] y)
         public static void MemoryCopy(byte[] src, int srcOffset, byte[] dst, int dstOffset, int count)
         {
             if (count == 0) return;
+            if (srcOffset < 0 || count < 0 || dstOffset < 0) throw new ArgumentException();
+            if (checked(srcOffset + count) > src.Length) throw new ArgumentException();
+            if (checked(dstOffset + count) > dst.Length) throw new ArgumentException();
```

Comment:
The `unsafe` methods shouldn't check these parameters. The responsibility for checking should be at the caller.